File: go/p6/p6_v1.go
package main

import (
	"fmt"
	"time"
)

func problem6() int {
	const firstNNumbers int = 100

	const sumOfSquares int = (firstNNumbers * (firstNNumbers + 1) * (2*firstNNumbers + 1)) / 6
	const squareOfSum int = (firstNNumbers * (firstNNumbers + 1) / 2) * (firstNNumbers * (firstNNumbers + 1) / 2)

	return squareOfSum - sumOfSquares
}

func main() {
	start := time.Now()
	fmt.Println("Result:", problem6())
	elapsed := time.Since(start)
	fmt.Printf("Execution time: %.6f seconds\n", elapsed.Seconds())
}

File: python/p6/p6_v1.py
def problem6():
    first_n_numbers = 100

    sum_of_squares = sum([i ** 2 for i in range(first_n_numbers + 1)])
    square_of_sum = sum([i for i in range(first_n_numbers + 1)]) ** 2

    return square_of_sum - sum_of_squares

if __name__ == "__main__":
    import time
    
    start_time = time.time()
    result = problem6()
    end_time = time.time()

    print(f"Result: {result}")
    print(f"Execution time: {end_time - start_time:.6f} seconds")